src/awx_plugins/interfaces/_temporary_private_api.py

```diff
@@ -4,6 +4,52 @@
 The hope is that it will be refactored into something more standardized.
 """
 
+from collections.abc import Callable
+
+
+class Credential:
+    """Input supplied by the user.
+
+    Satisfies ManagedCredentialType inputs want(s).
+    """
+
+    def __init__(self: 'Credential') -> None:
+        self.inputs: dict[str, bool | str | int | float | None] = {}
+
+    def get_input(
+            self: 'Credential',
+            field_name: str,
+            default: bool | str | int | float | None = None,
+    ) -> bool | str | int | float | None:
+        """Get the user supplied value for a given field.
+
+        Given the name of a field, return the user supplied value.
+
+        :param field_name: Input key to check if a value was supplied..
+        :type field_name: str
+        :param default: Value to return if a value was not supplied by
+            the user
+        :type default: bool | str | int | float | None
+        :returns: True if user supplied a value, False otherwise.
+        :rtype: bool
+        """
+        return self.inputs.get(field_name, default)
+
+    def has_input(self: 'Credential', field_name: str) -> bool:
+        """Check if user supplied a value for a given field.
+
+        Given the name of a field, return True of False as to if a value
+        was provided for that field.
+
+        :param field_name: Input key to check if a value was supplied..
+        :type field_name: str
+        :returns: True if user supplied a value, False otherwise.
+        :rtype: bool
+        """
+        return field_name in self.inputs and self.inputs[field_name] not in {
+            '', None,
+        }
```

Comment:
This can be simplified as
```suggestion
        return self.inputs.get(field_name, None) not in {'', None}
```